File: MCAPI/Models/Route.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MCAPI.Models
{
    public class Route
    {
        public Route()
        {

        }
        public Route(Guid routeID, Guid addressID, double distance, bool motorWay, bool ferry, bool toll, string created, Guid? userID)
        {
            RouteID = routeID;
            AddressID = addressID;
            Distance = distance;
            Motorway = motorWay;
            Ferry = ferry;
            Toll = toll;
            Created = created;
            UserID = userID;
        }
        [Key]
        public Guid RouteID { get; set; }
        [Required]
        public double Distance { get; set; }
        [Required]
        public bool Motorway { get; set; }
        [Required]
        public bool Ferry { get; set; }
        [Required]
        public bool Toll { get; set; }
        [Required]
        public string Created { get; set; }

        //ForeignKeys
        public Guid? UserID { get; set; }
        public User User { get; set; }
        public Guid AddressID { get; set; }
        public Address Address { get; set; }
        public List<Event> Events { get; set; }
    }
}

File: McWorld.Web/ClientApp/src/app/route/Models/Route.ts
import Address from '../Models/Address';
export default class Route {
  RouteID: number = undefined;
  Distance: number = 0;
  Motorway: boolean = false;
  Ferry: boolean = false;
  Toll: boolean = false;
  Created: string = undefined;
  UserID: number = undefined
  Addresses: Array<Address> = new Array<Address>(2);

  constructor() { }

  CreateRoute(distance: number, ferry: boolean, toll: boolean, motorway: boolean, startAddress: Address, endAddress: Address) {
    this.Distance = distance;
    this.Ferry = ferry;
    this.Toll = toll;
    this.Motorway = motorway;
    this.Created = undefined;
    this.RouteID = undefined;
    this.UserID = undefined;
    this.Addresses = [
      {
        AddressId: undefined,
        StreetName: startAddress.StreetName,
        StreetNumber: startAddress.StreetNumber,
        City: startAddress.City,
        Country: startAddress.Country,
        Zipcode: startAddress.Zipcode,
        Longitude: '00000000000',
        Latitude: '111111111111'
      },
      {
        AddressId: undefined,
        StreetName: endAddress.StreetName,
        StreetNumber: endAddress.StreetNumber,
        City: endAddress.City,
        Country: endAddress.Country,
        Zipcode: endAddress.Zipcode,
        Longitude: '2222222222222',
        Latitude: '33333333333333'
      }
    ];
  }
}
